src/lib/connectAxesToLayout.js

```diff
@@ -10,10 +10,37 @@ import {
   getAxisTitle,
 } from '../lib';
 
-function computeAxesOptions(axes, _) {
-  const options = [{label: _('All'), value: 'allaxes'}];
-  for (let i = 0; i < axes.length; i++) {
-    const ax = axes[i];
+function computeAxesOptions(axes, props) {
+  const _ = props.localize;
+  let filteredAxes = axes;
+
+  if (props.name === 'Range Slider') {
```

Comment:
I would rather have all of these conditions be inlined in `src/default_panels` both in principle (because we shouldn't have this much coupling to UI details in the framework) and for localization reasons (this will all fail if "Range Slider" is in French or something :)

Reply:
i.e. we should have a new prop called `axisFilter` which is run here